app/lib/shared/utils.dart

```diff
@@ -148,6 +148,50 @@ class DurationTracker extends LastNTracker<Duration> {
       };
 }
 
+/// Builds a sorted list of the top-k items using the provided comparator.
+///
+/// The algorithm uses a binary tree insertion, resulting in O(N * log(K)) comparison.
```

Comment:
I don't think this is a binary tree insertion, but insertion in a sorted list, complexity O(n*k) (the insertion step is linear in k)

Reply:
You could use an actual binary tree (though that comes with its own overhead)

If `k` is small enough, simple list insertion might be better

Reply:
Also, if we stick with the list, consider using https://pub.dev/documentation/collection/latest/collection/binarySearch.html . Binary search is known as being notoriously hard to implement correctly

Reply:
`binarySearch` in this case is not usable, as it returns `-1` when the searched item is not in the collection, while this algorithm needs the index where the next item should be inserted into. Furthermore, in most of the cases there is no equality, it won't be in the collection. However, with close review, I don't think the algorithm is conceptually the same, except for this specific goal.

I've checked `List.insert` + `List.removeLast()` (if the list is over `k`) and overall did perform worse.